packages/netlify-cms-core/src/reducers/mediaLibrary.js

```diff
@@ -127,6 +129,15 @@ const mediaLibrary = (state = Map(defaultState), action) => {
         map.set('isPersisting', false);
       });
     }
+    case ADD_MEDIA_FILES_TO_LIBRARY: {
+      const { mediaFiles } = action.payload;
+      let updatedFiles = differenceBy(state.get('files'), mediaFiles, 'path');
+      mediaFiles.forEach(file => {
+        const fileWithKey = { ...file, key: uuid() };
+        updatedFiles = [fileWithKey, ...updatedFiles];
```

Comment:
Won't something like so:
```
[...mediaFiles.map(file => ({ ...file, key: uuid() })), ...updatedFiles]
```
be sufficient here ?

Reply:
Good catch, I'll fix it.